cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -400,13 +400,13 @@ arrow::Status VeloxShuffleWriter::init() {
 arrow::Status VeloxShuffleWriter::initIpcWriteOptions() {
   auto& ipcWriteOptions = options_.ipc_write_options;
   if (options_.prefer_evict) {
-    ipcWriteOptions.memory_pool = options_.memory_pool.get();
+    ipcWriteOptions.memory_pool = options_.memory_pool;
   } else {
     if (!options_.ipc_memory_pool) {
-      auto ipcMemoryPool = std::make_shared<LargeMemoryPool>(options_.memory_pool.get());
-      options_.ipc_memory_pool = std::move(ipcMemoryPool);
+      auto ipcMemoryPool = new LargeMemoryPool(options_.memory_pool);
+      options_.ipc_memory_pool = ipcMemoryPool;
     }
```

Comment:
Changes here are tricky, current codes set ipc_memory_pool with memory_pool in JniWrapper.cc#ShuffleWriterJniWrapper_nativeMake, after check several related patches I assume LargeMemoryPool was not used.

We may direct use same arrow pool for shuffle writer and provide a config to decide whether to use LargeMemoryPool when initialize it in MemoryManager.

@marin-ma could you help review this change? thanks!

Reply:
Thanks for your advice. Let's add a configuration for it.

Some background: LargeMemoryPool was introduced to alleviate the occurrence of page fault, and it was only used for shuffle compression. But since the shuffle compression part has been refactored many times, we haven't done performance testing around the latest implementation, so it's not used for now.

Reply:
cc: @FelixYBW

Reply:
I suggest avoiding bringing this sort of change to the final version of the PR. Storing a static instance is an option if it's not in use but we don't decide to remove it either? @Yohahaha

Reply:
yeah, this is not the final version, I prefer to add a configuration for using LargeMemoryPool first, and then finish this patch.

Reply:
> I prefer to add a configuration for using LargeMemoryPool first

After take some time to read codes, I give up it because add an unused config just for keep unused code does not make sense for me, and it will bring lots of changes in VeloxMemoryManager.

I keep codes in VeloxShuffleWriter.cc but refactor a little, please help review again, thanks! @marin-ma @zhztheplayer